src/tcn_ext/tcn_keys.rs

```diff
@@ -1,80 +1,169 @@
+                new_key
+            })
+    }
+
+    fn tck(&self) -> TemporaryContactKey {
+        self.preferences
+            .tck()
+            .map(|tck_bytes| Self::bytes_to_tck(tck_bytes))
+            .unwrap_or({ self.rak().initial_temporary_contact_key() })
+    }
+
+    fn set_tck(&self, tck: TemporaryContactKey) {
+        self.preferences.set_tck(Self::tck_to_bytes(tck));
+    }
+
+    fn rak_to_bytes(rak: ReportAuthorizationKey) -> [u8; 32] {
+        let mut buf = Vec::new();
+        rak.write(Cursor::new(&mut buf))
+            .expect("Couldn't write RAK bytes");
+        Self::byte_vec_to_32_byte_array(buf)
+    }
+
+    fn byte_vec_to_32_byte_array(bytes: Vec<u8>) -> [u8; 32] {
+        let mut array = [0; 32];
+        let bytes = &bytes[..array.len()]; // panics if not enough data
+        array.copy_from_slice(bytes);
+        array
+    }
+
+    fn byte_vec_to_tck_byte_array(bytes: Vec<u8>) -> TckBytesWrapper {
+        let mut array = [0; TCK_SIZE_IN_BYTES];
+        let bytes = &bytes[..array.len()]; // panics if not enough data
+        array.copy_from_slice(bytes);
+        TckBytesWrapper { tck_bytes: array }
+    }
+
+    pub fn bytes_to_rak(bytes: [u8; 32]) -> ReportAuthorizationKey {
+        ReportAuthorizationKey::read(Cursor::new(&bytes)).expect("Couldn't read RAK bytes")
+    }
+
+    pub fn tck_to_bytes(tck: TemporaryContactKey) -> TckBytesWrapper {
+        let mut buf = Vec::new();
+        tck.write(Cursor::new(&mut buf))
+            .expect("Couldn't write TCK bytes");
+        Self::byte_vec_to_tck_byte_array(buf)
+    }
+
+    fn bytes_to_tck(tck: TckBytesWrapper) -> TemporaryContactKey {
+        TemporaryContactKey::read(Cursor::new(&tck)).expect("Couldn't read TCK bytes")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_rak() {
+        let new_key = ReportAuthorizationKey::new(rand::thread_rng());
+        // println!("{}", new_key);
+
+        let bytes = TcnKeysImpl::<PreferencesMock>::rak_to_bytes(new_key);
+        println!("{:?}", bytes);
+        assert_eq!(1, 1);
+    }
+
+    #[test]
+    fn test_load_rak() {
+        let bytes = [
+            42, 118, 64, 131, 236, 36, 122, 23, 13, 108, 73, 171, 102, 145, 66, 91, 157, 105, 195,
+            126, 139, 162, 15, 31, 0, 22, 31, 230, 242, 241, 225, 85,
+        ];
+        let key = TcnKeysImpl::<PreferencesMock>::bytes_to_rak(bytes);
+
+        let tck = key.initial_temporary_contact_key();
+
+        println!("tck initial: {:#?}", tck);
+
+        let tck_bytes = TcnKeysImpl::<PreferencesMock>::tck_to_bytes(tck);
+
+        println!("Bytes: {:#?}", tck_bytes);
+    }
+
+    #[test]
+    fn test_load_tck() {
+        let rak_bytes = [
+            42, 118, 64, 131, 236, 36, 122, 23, 13, 108, 73, 171, 102, 145, 66, 91, 157, 105, 195,
+            126, 139, 162, 15, 31, 0, 22, 31, 230, 242, 241, 225, 85,
+        ];
+        let rak = TcnKeysImpl::<PreferencesMock>::bytes_to_rak(rak_bytes);
```

Comment:
Maybe add a convenience initializer/associated function to RAK, to init with bytes (like e.g. `SignedReportExt:: with_str` added in https://github.com/Co-Epi/app-backend-rust/pull/31 ?)